core/cautils/localgitrepository.go

```diff
@@ -122,82 +111,65 @@ func (g *LocalGitRepository) GetLastCommit() (*apis.Commit, error) {
 	}, nil
 }
 
-func (g *LocalGitRepository) getAllCommits() ([]*git2go.Commit, error) {
-	logItr, itrErr := g.git2GoRepo.Walk()
-	if itrErr != nil {
-
-		return nil, itrErr
+func (g *LocalGitRepository) getAllCommits() ([]*objectv5.Commit, error) {
+	ref, err := g.goGitRepo.Head()
+	if err != nil {
+		return nil, err
 	}
-
-	pushErr := logItr.PushHead()
-	if pushErr != nil {
-		return nil, pushErr
+	logItr, err := g.goGitRepo.Log(&gitv5.LogOptions{From: ref.Hash()})
+	if err != nil {
+		return nil, err
 	}
 
-	var allCommits []*git2go.Commit
-	err := logItr.Iterate(func(commit *git2go.Commit) bool {
+	var allCommits []*objectv5.Commit
+	err = logItr.ForEach(func(commit *objectv5.Commit) error {
```

Comment:
@matthyx so if I understand correctly, this is the critical part that used to be slow (and might still be) and justified the use of the native git bindings? Is that correct?

Reply:
check the other comment